opentelemetry/src/main/scala/zio/telemetry/opentelemetry/PropagatingSupervisor.scala

```diff
@@ -0,0 +1,45 @@
+package zio.telemetry.opentelemetry
+
+import io.opentelemetry.api.trace.Span
+import io.opentelemetry.context.Context
+import zio._
+
+import java.util.concurrent.ConcurrentHashMap
```

Comment:
I guess it can be replaced with https://zio.dev/reference/sync/concurrentmap. As a consequence, you will have `Supervisor[UIO[Unit]]` and `@nowarn` annotation might be omitted.

Reply:
I'm afraid we'll have to stick to the regular `ConcurrentHashMap` because all callbacks in `Supervisor` are [unsafe and return `Unit`](https://github.com/zio/zio/blob/a9c924d4d346c3561e298546236e15b29d3b72c4/core/shared/src/main/scala/zio/Supervisor.scala#L53-L66).

Reply:
`@nowarn` basically tells us that we are throwing away `Scope` that we get when we call `Context.makeCurrent()`. Ideally we shouldn't do this. I saw that other tracing libraries (like Datadog's tracing SDK) use stacks of scopes. So potentially we could change `storage` to be of type `ConcurrentHashMap[FiberId, Stack[Span]]`. But this is kind of involved and apparently this all works even when we do throw scopes away.